runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -79,29 +82,47 @@ public void handleFailure(IReconfigurationHandlerPolicy failureHandlerPolicy,
 
     /**
      * Takes in the existing layout and a set of failed nodes.
-     * It first generates a new layout by removing the failed nodes from the existing layout.
-     * It then seals the epoch to prevent any client from accessing the stale layout.
-     * Finally we run paxos to update all servers with the new layout.
+     * Dispatches a workflow request to heal the node.
      *
      * @param healingHandlerPolicy Policy dictating healing of the reviving node.
      * @param currentLayout        The current layout
-     * @param healedServers        Set of healed server addresses
+     * @param healedServer         Healed server address
      * @throws QuorumUnreachableException  if reconfiguration not accepted by quorum.
-     * @throws OutrankedException          if reconfiguration consensus attempt outranked.
      * @throws LayoutModificationException if attempting to reconfigure with invalid layout.
      */
     public void handleHealing(IReconfigurationHandlerPolicy healingHandlerPolicy,
                               Layout currentLayout,
-                              Set<String> healedServers)
-            throws QuorumUnreachableException, OutrankedException, LayoutModificationException {
+                              String healedServer) {
+        // The healNodeWorkflow request is sent to the tail of the log unit chain to optimize the
+        // time taken to read and replicate the data.
+        List<String> logServers = currentLayout.getSegments().get(0).getStripes().get(0)
+                .getLogServers();
+        String server = logServers.get(logServers.size() - 1);
+
+        CreateWorkflowResponse resp = runtime.getRouter(server).getClient(ManagementClient.class)
+                .healNodeRequest(healedServer, true, true, true, 0);
+        UUID workflowId = resp.getWorkflowId();
+
+        final long retryQueryTimeout = 500;
```

Comment:
Probably needs to be a parameter of the method

Reply:
Done.